src/core/pointcloud/qgscopcpointcloudindex.cpp

```diff
@@ -423,41 +419,48 @@ void QgsCopcPointCloudIndex::populateHierarchy( const char *hierarchyPageData, u
   for ( uint64_t i = 0; i < byteSize; i += sizeof( CopcEntry ) )
   {
     const CopcEntry *entry = reinterpret_cast<const CopcEntry *>( hierarchyPageData + i );
-    const IndexedPointCloudNode nodeId( entry->key.level, entry->key.x, entry->key.y, entry->key.z );
+    const QgsPointCloudNodeId nodeId( entry->key.level, entry->key.x, entry->key.y, entry->key.z );
     mHierarchy[nodeId] = entry->pointCount;
     mHierarchyNodePos.insert( nodeId, QPair<uint64_t, int32_t>( entry->offset, entry->byteSize ) );
   }
 }
 
-bool QgsCopcPointCloudIndex::hasNode( const IndexedPointCloudNode &n ) const
+bool QgsCopcPointCloudIndex::hasNode( const QgsPointCloudNodeId &n ) const
 {
   return fetchNodeHierarchy( n );
 }
 
-QList<IndexedPointCloudNode> QgsCopcPointCloudIndex::nodeChildren( const IndexedPointCloudNode &n ) const
+QgsPointCloudNode QgsCopcPointCloudIndex::getNode( const QgsPointCloudNodeId &id ) const
 {
-  fetchNodeHierarchy( n );
+  Q_ASSERT( fetchNodeHierarchy( id ) );
```

Comment:
```suggestion
  bool nodeFound = fetchNodeHierarchy( id );
  Q_ASSERT( nodeFound );
```
@dvdkon I'm worried this line could end up with no-op in release builds - and Qt docs seem to confirm that: https://doc.qt.io/qt-6/debug.html

"Q_ASSERT(), Q_ASSERT_X(), and Q_CHECK_PTR() expand to nothing if QT_NO_DEBUG is defined during compilation. For this reason, the arguments to these macro should not have any side-effects."

Reply:
You're right, this is a problem, thanks for catching it. What's strange is that none of the CI tests caught it, they should build QGIS in [release mode](https://github.com/qgis/QGIS/actions/runs/12122173381/job/33794754839?pr=59564#step:10:123).

Reply:
I just tried removing the call in question and all tests still pass, they still get the right `pointCount` numbers. I don't know exactly why, but I suspect that somehow the exact sequence of accesses ends up fetching the nodes in question before `getNode` is called.